app/controllers/shf_applications_controller.rb

```diff
@@ -218,6 +218,7 @@ def new_file_uploaded(params)
                                            filename: @uploaded_file.actual_file_file_name))
           successful = successful & true
         else
+          @shf_application.uploaded_files.delete(@uploaded_file)
```

Comment:
@patmbolger I deleted the object because even if the `@uploaded_file` was invalid (here we are in the `else` statement for the `@uploaded_file.valid?` validation) it was hanging there as part of the uploaded files when we were back at the view.

Reply:
No problem - that's fine doing it here.